src/components/video/SimpleVideo.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import { Animated, StyleSheet, View, ViewStyle } from 'react-native';
+// @ts-ignore
+import Sound from 'react-native-sound';
+// @ts-ignore
+import Video from 'react-native-video';
+import convertToProxyURL from 'react-native-video-cache';
+import styled from 'styled-components';
+import { useTheme } from '../../context/ThemeContext';
+import { useAudio } from '@rainbow-me/hooks';
+import { ImgixImage } from '@rainbow-me/images';
+import { position } from '@rainbow-me/styles';
+import logger from 'logger';
+
+export type SimpleVideoProps = {
+  readonly style?: ViewStyle;
+  readonly uri: string;
+  readonly posterUri?: string;
+  readonly loading: boolean;
+  readonly setLoading: (isLoading: boolean) => void;
+};
+
+const absoluteFill = `
+  position: absolute;
+  ${position.size('100%')}
+`;
+
+const StyledBackground = styled(View)`
+  ${absoluteFill}
+`;
+
+const StyledVideo = styled(Video)`
+  ${absoluteFill}
+`;
+
+const StyledPosterContainer = styled(Animated.View)`
+  ${absoluteFill}
+`;
+
+const StyledImgixImage = styled(ImgixImage)`
+  ${position.size('100%')}
+`;
+
+const styles = StyleSheet.create({
+  flex: { flex: 1 },
+});
+
+export default function SimpleVideo({
+  style,
+  uri,
+  posterUri,
+  loading,
+  setLoading,
+}: SimpleVideoProps): JSX.Element {
+  const { colors } = useTheme();
+  const { white } = colors;
+  const ref = React.useRef<Video>();
+  const source = React.useMemo(
+    () => ({
+      uri: convertToProxyURL(uri),
+    }),
+    [uri]
+  );
+  const { currentSound, isPlayingAsset, fadeTo } = useAudio();
+  const [opacity] = React.useState<Animated.Value>(
+    () => new Animated.Value(loading ? 1 : 0)
+  );
+
+  const [soundOnMount] = React.useState<Sound | null>(
+    isPlayingAsset && currentSound ? currentSound : null
+  );
+
+  React.useEffect(() => {
+    Animated.timing(opacity, {
+      duration: 1000,
+      toValue: loading ? 1 : 0,
+      useNativeDriver: true,
+    }).start();
+  }, [opacity, loading]);
+
+  // XXX: Force the player to quit when unmounted. (iOS)
+  React.useEffect(() => {
+    const { current } = ref;
+    !!soundOnMount &&
+      (async () => {
```

Comment:
I don't understand. Why this is `async` and actually evaluated just in place and even not having any `await`...?

Reply:
I generally do this for two reasons:
  - I lean towards unhandled rejections over crashes
  - I try to treat global state transitions as asynchronous actions to enforce a clear separation between events + consuming state

But this is all just for development. Once we're happy with a certain pattern it makes perfect sense to remove these. +1